src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -105,37 +105,43 @@ AttributeSyntax attribute
 	private ArrowExpressionClauseSyntax? Transform(
 		ArrowExpressionClauseSyntax? body
 	) {
-		if( body == null ) {
+		if( body is null ) {
 			return null;
 		}
 
-		GeneratorError(
-			body.GetLocation(),
-			"ArrowExpressionClauseSyntax is not supported"
-		);
-
-		return body;
+		return body.WithExpression( Transform( body.Expression ) );
 	}
 
 	private BlockSyntax? Transform( BlockSyntax? block ) {
-		if( block == null ) {
+		if( block is null ) {
 			return null;
 		}
 
-		// { throw null; }
-		return SyntaxFactory.Block(
-			SyntaxFactory.ThrowStatement(
-				SyntaxFactory.Token( SyntaxKind.ThrowKeyword )
-					.WithTrailingTrivia( SyntaxFactory.Space ),
-				SyntaxFactory.LiteralExpression(
-					SyntaxKind.NullLiteralExpression,
-					SyntaxFactory.Token( SyntaxKind.NullKeyword )
-				),
-				SyntaxFactory.Token( SyntaxKind.SemicolonToken )
-			)
-		).WithTriviaFrom( block );
+		var transformed = TransformAll( block.Statements, Transform );
+
+		return block.WithStatements( transformed );
 	}
 
+	private StatementSyntax Transform( StatementSyntax stmt )
+		=> stmt switch {
+			ExpressionStatementSyntax exprStmt =>
+				exprStmt.WithExpression( Transform( exprStmt.Expression) ),
+
+			ReturnStatementSyntax returnStmt =>
+				returnStmt.Expression is null
+				? returnStmt
+				: returnStmt.WithExpression( Transform( returnStmt.Expression ) ),

```

Comment:
Most things will look like this: handle a case by recursing and adding the result back in with WithX.

In practice I found very few things would need more intelligent handling. There is already support for `Task<T>` -> `T` in this file, the real big thing is around await expressions and some interesting stuff with async Lambdas (if we choose to eventually support that, even).

Reply:
```cs
ReturnStatementSyntax returnStmt => returnStmt
    .WithExpression( MaybeTransform( returnStmt.Expression, Transform ) ),